### PR TITLE
feat!: Add invalid prop to Field Set and update Field and Field Set docs

### PR DESCRIPTION
**Pull request by alimpens**

### CODE REVIEW

File: packages/react/src/Fieldset/Fieldset.tsx
Comment (VincentSmedinga) [inline packages/react/src/Fieldset/Fieldset.tsx:1]: Separating breaking changes from non-breaking ones would help our implementers find the exact changes they need to make when upgrading. Is it worthwhile to extract a separate PR for the `FieldSet` rename? Is everything else non-breaking?

File: storybook/src/components/Field/Field.docs.mdx [hunk 1]
```diff
@@ -10,6 +10,14 @@ import README from "../../../../packages/css/src/components/field/README.md?raw"
 
 <Controls />
 
+## With Description
+
+A Field can have a description.
+Make sure to connect this description to the input in the Field,
+otherwise this won’t be read by a screen reader.
+
```
Comment (VincentSmedinga) [inline storybook/src/components/Field/Field.docs.mdx:18]: Let’s be explicit in the markup people need to add, rather than letting them find it in the Code View.

```suggestion
Add an `aria-describedby` attribute to the Field
and provide the `id` of the describing element as its value.

```

File: storybook/src/components/FieldSet/FieldSet.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+<Controls />
+
+## Examples
+
+## With Description
+
+A Field Set can have a description.
+Make sure to connect this description to the Field Set or a specific input,
+otherwise this won’t be read by a screen reader.
+
```
Comment (VincentSmedinga) [inline storybook/src/components/FieldSet/FieldSet.docs.mdx:20]:
```suggestion
Add an `aria-describedby` attribute to the Field Set
and provide the `id` of the describing element as its value.

```

File: storybook/src/components/Field/Field.docs.mdx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/src/components/Field/Field.docs.mdx:18]: I see you’ve already did so with the examples for radio and checkbox.

File: storybook/src/components/FieldSet/FieldSet.docs.mdx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Always also set `aria-required` on the individual radio buttons though, to make sure it’s read by screen readers.
+
+<Canvas of={FieldSetStories.RadioGroup} />
+
+### Checkbox group
+
+Use a Field Set to group checkboxes.
+
+Please note: [NVDA has bug](https://github.com/nvaccess/nvda/issues/12718) which causes it to
+not report a description connected to a Field Set when the Field Set contains checkboxes.
```
Comment (VincentSmedinga) [inline storybook/src/components/FieldSet/FieldSet.docs.mdx:45]:
```suggestion
not report a description connected to a Field Set when it contains checkboxes.
```

File: storybook/src/components/FieldSet/FieldSet.docs.mdx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+A Field Set can indicate whether any of the inputs it contains has a validation error.
+
+<Canvas of={FieldSetStories.WithError} />
+
+### Radio group
+
+Use a Field Set to group radio buttons.
+When grouping radio inputs, use `role="radiogroup"` on Field Set to have this grouping explicitly announced as a radio group.
+Field Set has a default role of `group`.
```
Comment (VincentSmedinga) [inline storybook/src/components/FieldSet/FieldSet.docs.mdx:33]:
```suggestion
When grouping radio inputs, use `role="radiogroup"` on Field Set to have this grouping explicitly announced as a radio group (the default role is `group`).
```

File: packages/react/src/Fieldset/Fieldset.tsx
Comment (alimpens) [reply packages/react/src/Fieldset/Fieldset.tsx:1]: Yeah, that's possible, and it would be clearer. However, doing case only renames is kind of horrible, especially on Windows machines. So if we can avoid it, that'd make me happy 😅

File: storybook/src/components/Field/Field.docs.mdx (hunk 1 above)
Comment (alimpens) [reply storybook/src/components/Field/Field.docs.mdx:18]: Your suggestion isn't quite correct, at least it isn't the current implementation.

```suggestion
Add an `aria-describedby` attribute to the input and provide the `id` of the describing element as its value.

```

File: packages/react/src/Fieldset/Fieldset.tsx
Comment (VincentSmedinga) [reply packages/react/src/Fieldset/Fieldset.tsx:1]: Next time then :)

File: storybook/src/components/Field/Field.docs.mdx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/src/components/Field/Field.docs.mdx:19]: Sorry, badly copied and pasted from Field Set.

**Review by VincentSmedinga — APPROVED** [review]
